src/scenario/custom_media.c

```diff
@@ -0,0 +1,93 @@
+#include "custom_media.h"
+
+#include "core/array.h"
+#include "core/log.h"
+
+#define CUSTOM_MEDIA_ARRAY_SIZE_STEP 100
+
+static array(custom_media_t) custom_media;
+
+static int entry_in_use(const custom_media_t *entry)
+{
+    return entry->type != CUSTOM_MEDIA_UNDEFINED;
+}
+
+static void new_entry(custom_media_t *obj, int position)
+{
+    obj->id = position;
+}
+
+void custom_media_clear(void)
+{
+    if (!array_init(custom_media, CUSTOM_MEDIA_ARRAY_SIZE_STEP, new_entry, entry_in_use) ||
+        !array_next(custom_media)) {
+        log_error("Unable to allocate enough memory for the custom media array. The game will now crash.", 0, 0);
+    }
+}
+
+custom_media_t *custom_media_get(int media_id)
+{
+    return array_item(custom_media, media_id);
+}
+
+custom_media_t *custom_media_create_blank(void)
+{
+    custom_media_t *entry = 0;
+    array_new_item(custom_media, 1, entry);
+    if (!entry) {
+        return 0;
+    }
+
+    return entry;
```

Comment:
```suggestion
    return entry;
```

You can replace this whole thing with a simple `return entry`.

Reply:
Done.